### PR TITLE
ipq40xx: add support for Sony NCP-HG100/Cellular

### PR DESCRIPTION
**Pull request by musashino205**
This patch series add support for Sony NCP-HG100/Cellular, include basic RGB LED support belonged to the micro-controler unit.

NCP-HG100 also has "/WLAN" variant with no WWAN module, but I don't have it and this patch series support "/Cellular" variant only.

### CODE REVIEW

File: target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Index: linux-5.4.98/drivers/leds/Kconfig
+===================================================================
+--- linux-5.4.98.orig/drivers/leds/Kconfig
++++ linux-5.4.98/drivers/leds/Kconfig
+@@ -823,6 +823,13 @@ config LEDS_LM36274
+ 	  Say Y to enable the LM36274 LED driver for TI LMU devices.
```
Comment (adschm) [inline target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch:6]: please, no 998- and 999- patches. Why would exactly you be the person who is last?

File: target/linux/ipq40xx/base-files/lib/upgrade/sony.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
```
Comment (adschm) [inline target/linux/ipq40xx/base-files/lib/upgrade/sony.sh:1]: No shebang for /lib files please (and no execute bit, I haven't checked if there is one ...)

Comment (musashino205) [reply target/linux/ipq40xx/base-files/lib/upgrade/sony.sh:1]: Thank you, I forgot it and I'll drop shebang. Currently, sony.sh has no execute bit and I think it's OK.

File: target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch (hunk 1 above)
Comment (musashino205) [reply target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch:6]: Thank you, I've forgotten to check the [rule](https://openwrt.org/docs/guide-developer/build-system/use-patches-with-buildsystem) of the number in the prefix, I'll change to ```8xx - other drivers```.

**Comment by musashino205** [comment]
UPDATE:

- drop shebang from sony.sh
- change the number of prefix to the range ```8xx``` for sony-mcu and leds-sony-mcu patches

and additional change:

- add detailed comment to sony.sh about ```rootfs_data``` partition usage

**Comment by musashino205** [comment]
UPDATE:

- rebase from master
  - adjust rows count in 901-arm-boot-add-dts-files.patch

File: target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch (hunk 1 above)
Comment (robimarko) [reply target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch]: This driver should be sent upstream first

Comment (musashino205) [reply target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch]: These drivers (sony-mcu, leds-sony-mcu) are very device-specific and not generic, but should be sent...?

Comment (robimarko) [reply target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch]: Yes, does not matter that they are device specific.
They should be upstreamed.

Comment (musashino205) [reply target/linux/ipq40xx/patches-5.4/999-leds-add-leds-sony-mcu-driver.patch]: I see... I'll rename drivers to the device-specific name and try to send to upstream.

**Comment by musashino205** [comment]
Temporary, I'll drop drivers from this PR to give priority to merging.
On stock firmware, "CLOUD" LED partially indicates system status, mainly related to the cloud-services. So use it instead of the RGB LED.
If the drivers are accepted in upstream, I'll re-add.

**Comment by musashino205** [comment]
UPDATE:

- rebase from master
  - adjust rows count in 901-arm-boot-add-dts-files.patch

**Comment by musashino205** [comment]
UPDATE:

- rebase from master

**Review by himalock — COMMENTED** [review]
It can be used only as a router.
Some features are not available. 
Front touch buttons and LEDs can't control, built-in bluetooth/sound speaker/z-wave, etc not working.
touch panel and LEDs could be with i2c-tools.

**Review by himalock — APPROVED** [review]

**Comment by musashino205** [comment]
I'll rebase from master.

**Comment by musashino205** [comment]
UPDATE:

- rebase from master

Note: This support uses ext4 format instead of F2FS, due to the too many errors if F2FS used
(full log: https://gist.github.com/musashino205/6e652cc198f28ffd8b9eb4ae290995da)

**Comment by musashino205** [comment]
I'll rebase and re-send to the ML, close.

**Pull request closed without merging**